src/InflateStream.php

```diff
@@ -20,10 +20,35 @@ class InflateStream implements StreamInterface
 
     public function __construct(StreamInterface $stream)
     {
-        // Skip the first 10 bytes
-        $stream = new LimitStream($stream, -1, 10);
+        // read the first 10 bytes, ie. gzip header
+        $header = $stream->read(10);
+        $filename_header_length = $this->getLengthOfPossibleFilenameHeader($stream, $header);
+        // Skip the header, that is 10 + length of filename + 1 (nil) bytes
+        $stream = new LimitStream($stream, -1, 10 + $filename_header_length);
         $resource = StreamWrapper::getResource($stream);
         stream_filter_append($resource, 'zlib.inflate', STREAM_FILTER_READ);
         $this->stream = new Stream($resource);
     }
+
+    /**
+     * @param StreamInterface $stream
+     * @param $header
+     * @return int
+     */
+    protected function getLengthOfPossibleFilenameHeader(StreamInterface $stream, $header)
+    {
+        $filename_header_length = 0;
+
+        if( substr(bin2hex($header), 6, 2) === '08' )
```

Comment:
Can you update this if statement to match PSR-2 coding standards? http://www.php-fig.org/psr/psr-2/#5-1-if-elseif-else